src/lib/rc/dsm.cpp

```diff
@@ -52,7 +52,7 @@
 #include "common_rc.h"
 #include <drivers/drv_hrt.h>
 
-#if defined (__PX4_LINUX) || defined (__PX4_DARWIN) || defined(__PX4_QURT)
+#if defined (__PX4_LINUX) || defined (__PX4_DARWIN) || defined(__PX4_CYGWIN) || defined(__PX4_QURT)
```

Comment:
Flip this around to handle the special nuttx case first, otherwise use usleep.

@davids5 do you know why nuttx uses a delay instead of sleep here?

Ideally rc could be kept as a standalone lib without PX4 specifics.

Reply:
Probably because it's used from IRQ context. Then it also makes sense to have the ifdef's in the current way: if we were to add support for another RTOS, we would stumble upon this, and realize that `usleep` cannot be used.

Reply:
I agree, then it's more readable and the chances for a new platform which might be added is higher to succeed without error.

Reply:
Ah right, the bind is done in the px4io serial dma callback.